Cython/Compiler/FusedNode.py

```diff
@@ -474,9 +479,20 @@ def _buffer_checks(self, buffer_types, pythran_types, pyx_code, decl_code, env):
         self._buffer_check_numpy_dtype(pyx_code, buffer_types, pythran_types)
         pyx_code.dedent(2)
 
-        for specialized_type in buffer_types:
-            self._buffer_parse_format_string_check(
-                    pyx_code, decl_code, specialized_type, env)
+        # creating a Cython memoryview from a Python memoryview is efficient
```

Comment:
This doesn't actually seem to be the case looking in more detail (although it probably could become the case). I suspect the optimization is mainly just in being able to skip creating the Cython memoryview much of the time